src/services/pcn-simplebridge/src/Simplebridge_dp.c

```diff
@@ -37,13 +37,23 @@ struct fwd_entry {
 } __attribute__((packed));
 
 BPF_TABLE("hash", __be64, struct fwd_entry, fwdtable, 1024);
+BPF_TABLE_SHARED("percpu_array", int, uint64_t, timestamp, 1);
```

Comment:
What about, if a control plane thread updates timestamp every second.
During the update lock will be per-cpu, not global.

But is also true that control plane thread updates all cpus values at the same time, by locking them all.
And it could be also matter of cache efficiency: if we have just an array instead of a percpu array.

I think it could be interesting to test if there is a performance difference with percpu/non-percpu versions.

Reply:
In the particular case of arrays there are not locks on the update. 

AFAIK percpu maps are important when the datapath updates elements, for example a counter, each cpu has its own and can update it without using atomic or synchronized operations. When the map is only read by datapath there is not any difference.